### PR TITLE
azdevify chat sample

### PR DESCRIPTION
**Pull request by v-jiaodi**
Azdevify this chat sample. 

**Finished:**
Create the required resources with `/infra`, and the resources were successfully created. Run `azd deploy`, the apps can be deployed normally.

**Questions:**
Use `containerapp` to deploy, it cannot achieve complete interaction.

Besides, we have tried use `appservice` to deploy apps, but the app startup failed.
The relevant code is here: https://github.com/v-jiaodi/chat-with-your-data-solution-accelerator/tree/convert-azd-with-appservice

@jongio , @ruoccofabrizio and @gmndrg Do you have any ideas on these two issues?

### CODE REVIEW

**Comment by jongio** [comment]
Can you include information about why it fails to start?  What are the symptoms?

File: infra/core/host/function.bicep [hunk 1]
```diff
@@ -0,0 +1,127 @@
+metadata description = 'Creates an Azure Function in an existing Azure App Service plan.'
```
Comment (jongio) [inline infra/core/host/function.bicep:1]: Did you copy this from latest?  The latest is functions.bicep, not function.bicep, so I'm a little concerned that core wasn't copied from the latest.  Can you double check all files are latest?

image

https://github.com/Azure/azure-dev/tree/main/templates/common/infra/bicep/core/host

File: infra/main.parameters.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
```
Comment (jongio) [inline infra/main.parameters.json:1]: Please use .bicepparam instead of mian.parameters.json

File: README.md [hunk 3]
```diff
@@ -104,6 +104,96 @@ Out-of-the-box, you can upload the following file types:
     
     ![A screenshot of the chat app.](./media/chat-app.png)
 
+## Develoy with Azure Developer CLI
```
Comment (jongio) [inline README.md:107]: Please check spelling and grammar

File: infra/app/web.bicep [hunk 4]
```diff
@@ -0,0 +1,52 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param identityName string
+param env array
+// param applicationInsightsName string
```
Comment (jongio) [inline infra/app/web.bicep:7]: Why comment out?

File: infra/core/host/function.bicep [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    applicationInsightsName: applicationInsightsName
+    appServicePlanId: appServicePlanId
+    AzureCognitiveSearch :AzureCognitiveSearch
+    formRecognizerName:FormRecognizerName
+    ContentSafetyName:ContentSafetyName
+    storageAccountName:storageAccountName
+    openAiName:openAiName
+    appSettings: union(appSettings, {
+        AzureWebJobsStorage: 'DefaultEndpointsProtocol=https;AccountName=${storage.name};AccountKey=${storage.listKeys().keys[0].value};EndpointSuffix=${environment().suffixes.storage}'
+        // AZURE_BLOB_ACCOUNT_KEY : storage.listKey().keys[0].value
```
Comment (jongio) [inline infra/core/host/function.bicep:68]: Please don't change /infra/core at all.  It should be a clean copy and paste from the latest.

File: infra/core/host/appservice.bicep [hunk 6]
```diff
@@ -0,0 +1,113 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+// Reference Properties
+param applicationInsightsName string = ''
+param appServicePlanId string
+param keyVaultName string = ''
+param managedIdentity bool = !empty(keyVaultName)
+param AzureCognitiveSearch string
```
Comment (jongio) [inline infra/core/host/appservice.bicep:10]: These settings shouldn't be in core, but in /app.

Please keep consistent casing. Camel for params.

**Review by jongio — COMMENTED** [review]
Please see my comments and update the bicep to match latest core and move app specific settings to /app.  Thanks

**Comment by zedy-wj** [comment]
@jongio We are currently investigating whether this chat sample uses `appservice` or `containerapp`, and we are currently experiencing some issues with both.

- For `appservice`: We used `js appservice host` when deploying the `frontend `module. The app startup failed, because it cannot match the `ts` module in the code. Currently, we use `ts staticwebapp host` and find that it can run normally. We will try to deploy other `services` as `appservice` later.

- For `containerapp`: Currently, this PR uses container app. The app can be deployed normally, but there is no data on front-end and back-end interactions.
 
Which one do you think is appropriate? At present, after investigation, we think that `appservice` is more suitable and are trying to modify it.
 
Regarding the comments on this PRs, if we decide to use the `containerapp`, we will resolve these comments ASAP .

**Comment by jongio** [comment]
> Which one do you think is appropriate?

Please follow what is here: https://github.com/jongio/chat-with-your-data-solution-accelerator/blob/main/infrastructure/deployment.bicep

**Comment by v-xuto** [comment]
@jongio Sorry, we no longer use this PR, and we will close this PR. We have created a new PR https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2 to replace it, and PR https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2 is ready, please review.

**Pull request closed without merging**